src/pwiz/data/msdata/Serializer_mzML.cpp

```diff
@@ -167,12 +167,19 @@ void Serializer_mzML::Impl::write(ostream& os, const MSData& msd,
     {
         stream_offset indexListOffset = xmlWriter.positionNext();
 
-        XMLWriter::Attributes attributes; 
-        attributes.push_back(make_pair("count", "2"));
+        XMLWriter::Attributes attributes;
```

Comment:
These changes ensure that an index element entry is only generated, if also the respective data is present in the file, i.e. a chromatogram index element is only added, if the file contains chromatographic data, a spectrum index element if it contains spectra.